playground/src/pages/CodingEnvironment.tsx

```diff
@@ -214,9 +232,25 @@ Cycles = [${trace_length.toString()}]`);
       toast.error("Not yet available");
     });
   };
+
+  /**
+   * This verifies the proof that is stored in the session.
+   * It runs the Rust program that is imported above.
+   * It returns true if the proof is valid, and false otherwise.
+   * As inputs we need program_info, stack_input, stack_output, and proof.
+   */
   const verifyProgram = async () => {
     init().then(() => {
-      toast.error("Not yet available");
+      try {
+        if (checkInputs(inputs) && checkOutputs(outputs, proof)) {
+          const result = verify_program(code, inputs, outputs, proof);
+          setOutput(`Verification succeeded with a security level of ${result} bits. \n \n \n`);
+          toast.success("Verification successful");
+        }
+      } catch (error) {
+          setOutput("Error: Check the developer console for details.");
```

Comment:
Understand that point. I think we don't get the error here from Rust atm. If the verification fails, we get `unreachable`. There might be a way to propagate the error as well. I can include that into the PR or we can as well work on that in the next UX PR

Reply:
Just so I understand - the error in the console is coming from the wasm package and you don't have access to it? If that's correct, then yes, we could tackle that as its own issue. I don't think we should have a single "UX PR". I think we'll want to break it down into a bunch of small steps, and this could be one of them

Reply:
You are correct. I don't have access to the wasm error (yet). I am sure somehow we can propagate it. 

And yes, we should have many small UX PRs